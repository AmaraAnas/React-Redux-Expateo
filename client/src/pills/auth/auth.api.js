import Api from '../api/base.api';

import {
  getNavigator,
  getResolution,
  isMobile,
  setSession,
  getSession,
} from '../../utils';

/*
export async function login(username, password) {
  let data = {
    ajaxAction: 'connect',
    gNavigator: getNavigator(),
    gResolution: getResolution(),
    gDevice: isMobile() ? 'M' : 'D',
    gApp: 'PG',
    USR_LANGUAGE: navigator.language.split('-')[0],
    USR_APP: 'XPTO',
    USR_REMEMBERME: true,
    USR_EMAIL: username,
    USR_PASSWORD: password,
  };

  return new Promise((resolve, reject) => {
    setTimeout(() => {
      Api.post('/ws/ajax/ajax_usr.php', data)
        .then(({ data }) => {
          let user = { ...data, isLogged: true };
          setSession(user);
          resolve(user);
        })
        .catch(reject);
    }, 1250);
  });
}

export async function loginAfterInscription(guid, familyid) {
  let data = {
    ajaxAction: 'connect',
    gNavigator: getNavigator(),
    gResolution: getResolution(),
    gDevice: isMobile() ? 'M' : 'D',
    gApp: 'XPTO',
    USR_LANGUAGE: navigator.language.split('-')[0],
    USR_APP: 'PG',
    USR_REMEMBERME_ID: guid,
    USR_REMEMBERME_FAMILY: familyid,
  };

  return new Promise((resolve, reject) => {
    setTimeout(() => {
      Api.post('/ws/ajax/ajax_usr.php', data)
        .then(({ data }) => {
          let user = { ...data, isLogged: true };
          setSession(user);
          resolve(user);
        })
        .catch(reject);
    }, 1250);
  });
}

export async function checkAuth(token) {
  let user = getSession();
  if (!user) {
    return Promise.resolve();
  }
  let data = {
    ajaxAction: 'connect',
    gNavigator: getNavigator(),
    gResolution: getResolution(),
    gDevice: isMobile() ? 'M' : 'D',
    gApp: 'PG',
    USR_LANGUAGE: navigator.language.split('-')[0],
    USR_APP: 'XPTO',
    USR_REMEMBERME: true,
    USR_REMEMBERME_EMAIL: user.gUsrEmail,
    USR_REMEMBERME_ID: user.gUsrGuid,
  };
  return Api.post('/ws/ajax/ajax_usr.php', data).then(({ data }) => {
    let user = { ...data, isLogged: true };
    setSession(user);
    return user;
  });
}
*/
export async function genericLogin(credentials, RememberMe, afterInscription) {
  let data = {
    ajaxAction: 'connect',
    gNavigator: getNavigator(),
    gResolution: getResolution(),
    gDevice: isMobile() ? 'M' : 'D',
    gApp: 'PG',
    USR_LANGUAGE: navigator.language.split('-')[0],
    USR_APP: 'XPTO',
    USR_REMEMBERME: true,
  };

  if (RememberMe) {
    let user = getSession();
    if (!user) {
      return Promise.resolve();
    } else {
      data['USR_REMEMBERME_EMAIL'] = user.gUsrEmail || '';
      data['USR_REMEMBERME_ID'] = user.gUsrGuid || '';
    }
  } else if (afterInscription) {
    data['USR_REMEMBERME_ID'] = credentials.guid || '';
    data['USR_REMEMBERME_FAMILY'] = credentials.family || '';
  } else {
    data['USR_EMAIL'] = credentials.username || '';
    data['USR_PASSWORD'] = credentials.password || '';
  }

  return new Promise((resolve, reject) => {
    setTimeout(() => {
      Api.post('/ws/ajax/ajax_usr.php', data)
        .then(({ data }) => {
          let user = { ...data, isLogged: true };
          setSession(user);
          resolve(user);
        })
        .catch(reject);
    }, 1250);
  });
}
