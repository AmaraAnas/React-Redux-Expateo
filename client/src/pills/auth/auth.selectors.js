// TODO: test it
export const userSelector = ({ Auth }) => Auth.user;
