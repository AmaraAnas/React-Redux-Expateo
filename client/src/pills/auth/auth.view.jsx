import React from 'react';
import { reduxForm, Field } from 'redux-form';
import PropTypes from 'prop-types';

import { Button, Form, Icon } from '../../ui-kit';
import { Input } from '../../redux-form-utils/fieldComponents';

function AuthView({ handleSubmit }) {
  return (
    <Form size="large" onSubmit={handleSubmit}>
      <Field
        name="email"
        component={Input}
        icon="user"
        iconPosition="left"
        placeholder="Email"
        autoFocus
        fluid
      />
      <Field
        name="password"
        type="password"
        component={Input}
        icon="lock"
        iconPosition="left"
        placeholder="Password"
        fluid
      />
      <Button primary type="submit" fluid size="large">
        S'identifier
      </Button>
    </Form>
  );
}

AuthView.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

export default reduxForm({
  form: 'AuthForm',
})(AuthView);
