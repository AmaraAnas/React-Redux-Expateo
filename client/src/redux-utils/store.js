import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { isFSA } from 'flux-standard-action';
import LogRocket from 'logrocket';
import setupLogRocketReact from 'logrocket-react';
import throttle from 'lodash.throttle';
import { routerMiddleware } from 'connected-react-router';

import * as themesApi from '../pills/themes/themes.api';
import * as servicesApi from '../pills/services/services.api';
import * as tasksApi from '../pills/tasks/tasks.api';
import * as mobilitiesApi from '../pills/mobilities/mobilities.api';

import reducers, { history } from './reducers';
import { loadState, saveState } from './localStorage';

const middlewares = [
  thunk.withExtraArgument({
    api: {
      themes: themesApi,
      services: servicesApi,
      tasks: tasksApi,
      mobilities: mobilitiesApi,
    },
  }),
  routerMiddleware(history),
];

// ---------------------------------------------------------------------
let composeEnhancers = compose;
if (process.env.NODE_ENV === 'development') {
  const fsaChecker = (store) => (next) => (action) => {
    if (isFSA(action)) {
      return next(action);
    } else {
      console.error('Dispatching a non FSA action => ', action);
      throw new Error(
        `Dispatching a non FSA action => ${JSON.stringify(action)}`,
      );
    }
  };
  middlewares.push(fsaChecker);
  middlewares.push(logger);
  composeEnhancers =
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || composeEnhancers;
}
// ---------------------------------------------------------------------
if (process.env.NODE_ENV === 'production') {
  LogRocket.init('1snrjg/develop-expateo-pa');
  //   LogRocket.identify('0123456789', {
  //     name: 'James Morrison',
  //     email: 'jamesmorrison@example.com',

  //     // Add your own custom user variables here, ie:
  //     subscriptionType: 'pro',
  //   });
  setupLogRocketReact(LogRocket);
  middlewares.push(LogRocket.reduxMiddleware());
}
// ---------------------------------------------------------------------
const persistedState = loadState();
const store = createStore(
  reducers,
  persistedState,
  composeEnhancers(applyMiddleware(...middlewares)),
);
// store.getState() is a save all. Maybe too much ?
store.subscribe(
  throttle(() => {
    const { Auth, router } = store.getState();
    saveState({ Auth, router });
  }, 1000),
);

export default store;
