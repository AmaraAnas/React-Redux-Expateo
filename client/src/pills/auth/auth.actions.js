import { createAction } from 'redux-actions';

import { addPrefixToActionTypes } from '../../redux-utils/utils';

import * as AuthApi from './auth.api';

export const ACTION_TYPES = addPrefixToActionTypes(
  {
    LOGIN_SUCCESS: 'LOGIN_SUCCESS',
    LOGIN_FAILURE: 'LOGIN_FAILURE',
  },
  'auth',
);

export const loginSuccess = createAction(ACTION_TYPES.LOGIN_SUCCESS);
const loginFailure = createAction(ACTION_TYPES.LOGIN_FAILURE);

export function login({
  username,
  password,
  onPending,
  onSuccess,
  onFailure,
  authApi = AuthApi,
}) {
  onPending();
  return async (dispatch) => {
    try {
      //const user = await authApi.login(username, password);
      const user = await authApi.genericLogin(
        { username, password },
        false,
        false,
      );
      dispatch(loginSuccess(user));
      onSuccess(user);
    } catch (e) {
      dispatch(loginFailure(e));
      onFailure(e);
    }
  };
}
