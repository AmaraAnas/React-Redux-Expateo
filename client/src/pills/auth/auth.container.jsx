import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import AuthViewForm from './auth.view';
import { login } from './auth.actions';
import { show, hide } from '../modal/modal.actions';
import { Loader } from '../modal/modal.container';

class AuthContainer extends Component {
  constructor(props) {
    super(props);
    this.handleLogin = this.handleLogin.bind(this);
  }

  handleLogin({ username, password }) {
    const { dispatch, onLogin } = this.props;
    const showModal = () => dispatch(show(Loader));
    const hideModal = () => dispatch(hide());
    dispatch(
      login({
        username,
        password,
        onPending: showModal,
        onSuccess: (user) => {
          hideModal();
          onLogin(user);
        },
        onFailure: (e) => {
          console.log('failure');
        },
      }),
    );
  }

  render() {
    return <AuthViewForm onSubmit={this.handleLogin} />;
  }
}

AuthContainer.propTypes = {
  onLogin: PropTypes.func.isRequired,
};

export default connect(null)(AuthContainer);
