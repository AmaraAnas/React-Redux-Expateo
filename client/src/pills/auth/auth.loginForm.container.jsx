import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { show, destroy } from '../modal/modal.actions';
import { BigLoaderModal } from '../modal/modal.loaders';

import AuthViewForm from './auth.loginForm.view';
import { login } from './auth.actions';

const showLoaderModal = () =>
  show(
    BigLoaderModal({
      content: 'Connexion en cours...',
    }),
  );

class AuthContainer extends Component {
  constructor(props) {
    super(props);
    this.handleLogin = this.handleLogin.bind(this);
  }

  handleLogin({ email, password }) {
    const { dispatch, onLogin } = this.props;
    const destroyModal = () => dispatch(destroy());
    dispatch(
      login({
        email,
        password,
        onPending: () => dispatch(showLoaderModal()),
        onSuccess: (user) => {
          destroyModal();
          onLogin(user);
        },
        onFailure: destroyModal,
      }),
    );
  }

  render() {
    return <AuthViewForm onSubmit={this.handleLogin} />;
  }
}

AuthContainer.propTypes = {
  onLogin: PropTypes.func.isRequired,
};

export default connect(null)(AuthContainer);
