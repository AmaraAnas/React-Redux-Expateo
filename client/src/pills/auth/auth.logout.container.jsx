import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { show, destroy } from '../modal/modal.actions';
import { BigLoaderModal } from '../modal/modal.loaders';

import { logout } from './auth.actions';

const showLoaderModal = () =>
  show(
    BigLoaderModal({
      content: 'DÃ©connexion en cours...',
    }),
  );

class LogoutContainer extends React.Component {
  constructor(props) {
    super(props);
    this.handleLogout = this.handleLogout.bind(this);
  }

  handleLogout() {
    const { dispatch } = this.props;
    dispatch(showLoaderModal());
    dispatch(logout());
    dispatch(destroy());
  }

  render() {
    return this.props.render({ logout: this.handleLogout });
  }
}

LogoutContainer.propTypes = {
  render: PropTypes.func.isRequired,
};

export default connect(null)(LogoutContainer);
