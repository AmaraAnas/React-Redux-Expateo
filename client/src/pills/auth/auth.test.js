import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import store from '../../redux-utils/store';
import AuthForm from './auth.loginForm.container';
import { login } from './auth.actions';
import AuthReducer from './auth.reducer';

describe('Auth render', () => {
  const noop = () => {};
  it('Should render', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <AuthForm onLogin={noop} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Auth action - reducers', () => {
  it('LOGIN Should call pending & success', () => {
    let isOnPendingCalledFlag = false;
    let isOnSuccessCalledFlag = false;
    let isOnFailureCalledFlag = false;
    const loginAction = login({
      username: 'jhon',
      password: 'doe',
      onPending: () => (isOnPendingCalledFlag = true),
      onSuccess: () => (isOnSuccessCalledFlag = true),
      onFailure: () => (isOnFailureCalledFlag = true),
      authApi: {
        classicLogin: () =>
          Promise.resolve().then(() => {
            expect(isOnPendingCalledFlag).toEqual(true);
            expect(isOnSuccessCalledFlag).toEqual(true);
            expect(isOnFailureCalledFlag).toEqual(false);
          }),
      },
    });
    loginAction(store.dispatch);
  });

  it('LOGIN Should call pending & failure', () => {
    let isOnPendingCalledFlag = false;
    let isOnSuccessCalledFlag = false;
    let isOnFailureCalledFlag = false;
    const loginAction = login({
      username: 'jhon',
      password: 'doe',
      onPending: () => (isOnPendingCalledFlag = true),
      onSuccess: () => (isOnSuccessCalledFlag = true),
      onFailure: () => (isOnFailureCalledFlag = true),
      authApi: {
        classicLogin: () =>
          Promise.reject().catch(() => {
            expect(isOnPendingCalledFlag).toEqual(true);
            expect(isOnSuccessCalledFlag).toEqual(false);
            expect(isOnFailureCalledFlag).toEqual(true);
          }),
      },
    });
    loginAction(store.dispatch);
  });

  it('LOGIN:SUCCESS Should set the user', () => {
    const noop = () => {};
    let state;
    const loginAction = login({
      username: 'jhon',
      password: 'doe',
      onPending: noop,
      onSuccess: noop,
      onFailure: noop,
      authApi: {
        classicLogin: () =>
          Promise.resolve({ username: 'jhon' }).then(() => {
            expect(state).toEqual({
              user: { username: 'jhon' },
              error: {
                message: '',
              },
            });
          }),
      },
    });
    state = AuthReducer(
      {
        user: {},
        error: {
          message: '',
        },
      },
      loginAction(store.dispatch),
    );
  });

  it('LOGIN:FAILURE Should set the error', () => {
    const noop = () => {};
    let state;
    const loginAction = login({
      username: 'jhon',
      password: 'doe',
      onPending: noop,
      onSuccess: noop,
      onFailure: noop,
      authApi: {
        classicLogin: () =>
          Promise.reject(new Error('rejected')).catch(() => {
            expect(state).toEqual({
              user: {},
              error: {
                message: 'rejected',
              },
            });
          }),
      },
    });
    state = AuthReducer(
      {
        user: {},
        error: {
          message: '',
        },
      },
      loginAction(store.dispatch),
    );
  });
});
