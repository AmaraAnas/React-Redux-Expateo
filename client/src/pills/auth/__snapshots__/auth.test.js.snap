// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth render Should render 1`] = `
<form
  className="ui large form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <div
      className="ui fluid left icon input"
    >
      <input
        autoFocus={true}
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
      <i
        aria-hidden="true"
        className="user icon"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="ui fluid left icon input"
    >
      <input
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <i
        aria-hidden="true"
        className="lock icon"
      />
    </div>
  </div>
  <button
    className="ui blue large fluid vertical animated button"
    onClick={[Function]}
    role="button"
    type="submit"
  >
    <div
      className="visible content"
    >
      S'identifier
    </div>
    <div
      className="hidden content"
    >
      <i
        aria-hidden="true"
        className="sign-in icon"
      />
    </div>
  </button>
</form>
`;
